ui/src/sources/containers/CreateSource.js

```diff
@@ -98,4 +115,4 @@ function mapStateToProps(_) {
   return {}
 }
 
-export default connect(mapStateToProps, {addSourceAction})(withRouter(CreateSource))
+export default connect(mapStateToProps, {addSourceAction, updateSourceAction})(withRouter(CreateSource))
```

Comment:
You can connect the `publishNotification` action to this component `ui/src/shared/actions/notifications.js` and do all the flashMessaging you desire!